src/main/java/uk/gov/cslearning/record/service/scheduler/LearningJob.java

```
@@ -64,6 +72,82 @@ public LearningJob(UserRecordService userRecordService, IdentityService identity
+            try {
+                Optional<CivilServant> optionalCivilServant = registryService.getCivilServantByUid(identity.getUid());
+
+                if (optionalCivilServant.isPresent()) {
+                    CivilServant civilServant = optionalCivilServant.get();
+                    List<Course> courses = learningCatalogueService.getRequiredCoursesByDepartmentCode(civilServant.getDepartmentCode());
+                    LOGGER.info("Got courses  ({})", courses.size());
+
+                    for (Course course : courses) {
+                        Collection<CourseRecord> courseRecords = userRecordService.getUserRecord(identity.getUid(), String.format(COURSE_URI_FORMAT, course.getId()));
+                        // okay we do not want to find a course without a completed record or without a record at all
+
+                        if (courseRecords.size() != 0) {
+                            for (CourseRecord courseRecord : courseRecords) {
+                                LOGGER.info("course completed  " + courseRecord.getCompletionDate());
+                                if (courseRecord.getCompletionDate() != null) {
+                                    LOGGER.info("course is COMPLETED");
+                                    try {
+                                        CheckAndNotifyLineManager(civilServant, identity, course, courseRecord.getCompletionDate());
+                                    } catch (NotificationClientException nce) {
+                                        LOGGER.error("Error notifying line manger about course completion");
+                                        throw nce;
+                                    }
+                                }
+                            }
+                        }
+                    }
+                } else {
+                    throw new HttpClientErrorException(HttpStatus.UNPROCESSABLE_ENTITY);
```

Just because an civil servant doesnt exist in CSRS, doesnt mean we should want to throw an exception here. Might it be worth just logging this and passing over?